### ISSUE TITLE
Micropython libraries

### ISSUE THREAD

**Issue by thupkens** [thonny/thonny#3018]
Cannot install **micropython** libraries from tools -> manage packages menu anymore. Worked well in version 4.0.2.
I use Windows 10 and Thonny 4.1.4.

**Comment by aivarannamaa** [comment]
> Cannot install micropython libraries from tools -> manage packages menu anymore. 

What does it mean precisely? Do you get an error message?

**Comment by pumpkinking10** [comment]
I have the same problem after installing numpy. Even after installing it using manage packages menu the program still shows error saying numpy is not recognised. I know that it was installed as it was shown on the <INSTALL> menu.
<img width="507" alt="image" src="https://github.com/thonny/thonny/assets/43786573/ab4a82d1-4dc5-4c87-8f98-6b82d557611e">
<img width="281" alt="image" src="https://github.com/thonny/thonny/assets/43786573/6d5ba1fb-ed7e-4fbe-9bbe-f5c5bed590eb">
The only thing that was off was that I had this message before installing numpy.
<img width="279" alt="image" src="https://github.com/thonny/thonny/assets/43786573/0ed3ceeb-8b50-419f-8cce-f8bd1b235415">

**Comment by pumpkinking10** [comment]
If it helps I am using thonny to control motor through a servo2040 controller. And I selected MicroPython(Raspberry Pi Pico) as the interpreter. This is selected when I used manage package to download numpy.

**Comment by aivarannamaa** [comment]
I'm surprised that you were able to install numpy on a MicroPython device, as it is quite large package and not meant for MicroPython.

**Comment by pumpkinking10** [comment]
Does this mean I should find a work around code and not use numpy? And does this mean it actually didn't installed numpy package? servo2040 controller have only 2MB so I guess that wouldn't be enough.

**Comment by aivarannamaa** [comment]
I'll check later what exactly happens when I try to install numpy on a device wit unsufficient flash.

There is a numpy alternative for small devices called ulab, but i don't know if it must be built into MicroPython or can be installed separately.

**Comment by pumpkinking10** [comment]
Yeah I am wondering if it can be installed separately as well because I am also using scipy.integrate.solve_ivp to integrate a function. I run the code in my computer instead of on servo2040 and it runs fine but when I switch the interpreter to Pi Pico it doesn't recognise numpy and scipy. I can get around numpy with ulab but I have no idea about scipy.integrate.

**Comment by pumpkinking10** [comment]
Actually, The servo2040 controller that I am using have 127MB storage so clearly it can house numpy. So I don't know why I can't use it after downloading it with Manage Package.

**Comment by Fabri54** [comment]
I've a problem when trying to instal a package from a location in my Mac.  If from "Manage Packages" I chose "Install from local file" and I click   "here"  Thonny crashes immediately and I can send a nice report to Apple.
Please note that  "Install from PyPi" works.
My Mac runs  Sonoma 14.1.2  on an M2 Pro chip
